src/test/java/marquez/db/DatasetDaoTest.java

```diff
@@ -77,13 +100,36 @@ public void testInsert() {
     assertThat(rowsAfter).isEqualTo(rowsBefore + 1);
   }
 
+  @Test(expected = UnableToExecuteStatementException.class)
+  public void testIdempotentInsert() {
```

Comment:
_Minor_: A better test name would be `testInsert_throwsException_onDuplicateRow`

Reply:
Yup :) Fixed.